recipes/coin-cgl/all/conanfile.py

```diff
@@ -32,30 +31,42 @@ def _source_subfolder(self):
     @property
     def _build_subfolder(self):
         return "build_subfolder"
+    
+    def export_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
-        if self.settings.os == "Windows" and self.options.shared:
-            raise ConanInvalidConfiguration("coin-cgl does not support shared builds on Windows")
         if self.options.shared:
             del self.options.fPIC
 
+    def validate(self):
+        if self.settings.os == "Windows" and self.options.shared:
+            raise ConanInvalidConfiguration("coin-cgl does not support shared builds on Windows")
+        if hasattr(self, "settings_build") and tools.cross_building(self):
```

Comment:
Maybe try same trick than https://github.com/conan-io/conan-center-index/pull/7162: add gnu-config in build requirements and override config.guess & config.sub. It may work only for static lib.